### PR TITLE
API /users v2

### PR DESCRIPTION
**Pull request by bensgilbert**
Slightly deviated from the spec on the `POST /users/{id}/experience` endpoint but hey. Would appreciate feedback (be critical please :))

### CODE REVIEW

File: api/src/routers/user.py [hunk 1]
```diff
@@ -1,3 +1,70 @@
...
+    UserCreatePublic,
+    UserPublic,
+    UserUpdateExperience,
+)
 
 router = APIRouter()
+
+
+@router.get("/user/{user_id}")
+async def get_user(
```
Comment (MilaDog) [inline api/src/routers/user.py:16]: Function docstring would be great.

File: api/src/routers/user.py [hunk 2]
```diff
@@ -1,3 +1,70 @@
...
+    session: Session = Depends(get_session),  # noqa: B008
+) -> UserPublic:
+    user = session.exec(select(User).where(User.user_id == user_id)).first()
+
+    if user is None:
+        raise HTTPException(404, "User not found")
+    return user
+
+
+@router.post("/user/{user_id}")
```
Comment (MilaDog) [inline api/src/routers/user.py:33]: POST to `/user` or event a `/user/create` would be the more ideal method. See https://github.com/FloncDev/exquisite-exoplanets/pull/32#discussion_r1685795869

File: api/src/routers/user.py [hunk 3]
```diff
@@ -1,3 +1,70 @@
...
+    session: Session = Depends(get_session),  # noqa: B008
+) -> Experience:
+    user = session.exec(select(User).where(User.user_id == user_id)).first()
+
+    if user is None:
+        raise HTTPException(404, "User not found")
+    return Experience(experience=user.experience)
+
+
+@router.post("/user/{user_id}/experience")
```
Comment (MilaDog) [inline api/src/routers/user.py:73]: PATCH is more ideal over POST.

**Review by MilaDog — CHANGES_REQUESTED** [review]
Looks good :D Just some small changes on the requets methods.

File: api/src/routers/user.py (hunk 3 above)
Comment (iauu) [reply api/src/routers/user.py:73]: Fixed in v3

File: api/src/routers/user.py (hunk 2 above)
Comment (bensgilbert) [reply api/src/routers/user.py:33]: Very true, something for v3!

**Review by MilaDog — APPROVED** [review]

**Pull request merged**